core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -759,12 +759,11 @@ class AdminClientIntegrationTest extends IntegrationTestHarness with Logging {
 
       val future = result.lowWatermarks().get(topicPartition)
       try {
-        lowWatermark = future.get(1000L, TimeUnit.MILLISECONDS).lowWatermark()
+        lowWatermark = future.get().lowWatermark()
         lowWatermark == 5L
       } catch {
-        case e: LeaderNotAvailableException => false
-      }
-
+        case e: ExecutionException if e.getCause.isInstanceOf[LeaderNotAvailableException] => false
```

Comment:
The test is failing when the cause is `NotLeaderForPartitionException`. Is that the exception that you meant to use? Or do we need to catch both?

Reply:
We've also seen it fail with LeaderNotAvailableException in the very beginning, so we probably need to catch both exceptions here. To be honest I never really questioned the original code that caught only LeaderNotAvailableException and simply fixed unnesting the exception.

Perhaps it would make sense to check for RetriableException instead?

Reply:
I've changed the code to test for a retriable exception and as far as I can tell it does what it is supposed to do. This should also cover all other relevant cases that allow us to resubmit the delete request I think. For anything else it is ok to fail the test I think.

I've also rebased and squashed.

Reply:
I think we should stick to checking the two leader related exceptions since those are the only ones we expect to be thrown in this case.

Reply:
I'm happy to change of course. I do wonder whether we might make the test too narrow though. If we encounter a retriable exception, retry and get the correct result, should we really fail the test?

Reply:
Yes, if we are throwing a retriable exception that is unexpected, the test should fail. We can then check if we made a mistake in the test or the code.